src/megameklab/com/printing/RecordSheetTask.java

```diff
@@ -177,36 +170,8 @@ protected String popupLabel() {
 
         @Override
         public Void doInBackground() throws Exception {
-            PDFMergerUtility merger = new PDFMergerUtility();
-            merger.setDestinationFileName(fileName);
-            Map<Integer, List<String>> bookmarkNames = new HashMap<>();
-            Iterator<PrintRecordSheet> iter = sheets.iterator();
-            while (iter.hasNext()) {
-                final PrintRecordSheet rs = iter.next();
-                bookmarkNames.put(rs.getFirstPage(), rs.getBookmarkNames());
-                for (int i = 0; i < rs.getPageCount(); i++) {
-                    merger.addSource(rs.exportPDF(i, pageFormat));
-                }
-                iter.remove();
```

Comment:
The sheet instances take up a lot more memory when processed, and large jobs can easily run out of memory. I tried clearing generated data from the PrintRecordSheet instance without success. The only way I found to handle the memory problems is to remove the sheets once they're added to the document. I don't see any reason the book needs to be reusable.

Reply:
I'll test this with a large document tonight, good to know that was the issue.

I found some other items that kept large amounts of memory rooted and will put up a separate PR.